framework/include/systems/NonlinearSystemBase.h

```diff
@@ -292,6 +292,8 @@ class NonlinearSystemBase : public SolverSystem, public PerfGraphInterface
 
   void setInitialSolution();
 
+  // Used for enforcing nodal BC's when using a direct acceleration calculation
+  void setNodalBCs();
```

Comment:
Can you elaborate on what you mean by a "passkey"?

Reply:
https://arne-mertz.de/2016/10/passkey-idiom/

Reply:
Got it, though the thumbs down from @GiudGiud is ominous...

Reply:
Maybe @GiudGiud would prefer that everyone be able to willy-nilly set nodal boundary conditions because our downstream applications have shown repeatedly that what you giveth they shall take. A general friending of all of `NonlinearSystemBase` would be an alternative but feels like overkill

Reply:
So for the passkey then, should I follow similar process of that found in `ReporterTransferInferface.c`? 

```
void
ReporterTransferInterface::addReporterTransferMode(const ReporterName & name,
                                                   const ReporterMode & mode,
                                                   FEProblemBase & problem)
{
  checkHasReporterValue(name, problem);
  problem.getReporterData(ReporterData::WriteKey())
      .getReporterStateBase(name)
      .addConsumer(mode, _rti_transfer);
}
```

Reply:
> A general friending of all of NonlinearSystemBase would be an alternative but feels like overkill

it would be simple and readable.
And it's not like people will start writing crazy calls into CentralDifference without our knowledge

but dont let me distract you

Reply:
Don't make David a pawn in your crusade against the passkey idiom! 😅

Reply:
David can chose the bright side of code simplicity by himself

Reply:
I'm inclined to agree with @GiudGiud. In my opinion, the passkey idiom harms readability. However my experience here is very limited.

Reply:
A lot of template code doesn't read very well but we use it anyway because it serves a functional purpose. The passkey gives very precise control and I would argue increases readability as it immediately makes clear exactly what `CentralDifference` needs access to in `NonlinearSystemBase`. Whereas if you just make it a `friend` you do not know what of `NonlinearSystemBase's` multitude of data it needs access to

Reply:
> as it immediately makes clear exactly what CentralDifference needs access to in NonlinearSystemBase

immediately after asking ChatGPT what's up with this struct key I've never seen before :D

Reply:
@lindsayad I took a shot at implementing a key
[a9736d1](https://github.com/idaholab/moose/pull/27991/commits/a9736d14fe42bdec55ae210f101518fb44977de5)

Reply:
I ask ChatGPT C++ questions on the reg. I do not mind